File: app/src/main/java/ua/com/motometer/android/core/facade/api/model/FuelRecord.java
package ua.com.motometer.android.core.facade.api.model;

import org.immutables.value.Value;

import java.math.BigDecimal;

@Value.Immutable
public interface FuelRecord extends Record {

    String petrolStationName();

    BigDecimal fuelPrice();

    BigDecimal fuelRefilled();
}

File: app/src/main/java/ua/com/motometer/android/core/facade/api/model/FuelRecord.kt
package ua.com.motometer.android.core.facade.api.model

import org.immutables.value.Value

import java.math.BigDecimal

@Value.Immutable
interface FuelRecord : Record {

    fun petrolStationName(): String

    fun fuelPrice(): BigDecimal

    fun fuelRefilled(): BigDecimal
}
